manim/mobject/table.py

```diff
@@ -604,10 +604,10 @@ def construct(self):
                 and self.col_labels is not None
                 and self.top_left_entry is None
             ):
-                index = len(self.mob_table) * (pos[0] - 1) + pos[1] - 2
+                index = self.col_dim * (pos[0] - 1) + pos[1] - 2
```

Comment:
```suggestion
                index = len(self.mob_table[0]) * (pos[0] - 1) + pos[1] - 2
```
Multiplying with `col_dim` doesn't work here because `col_dim` is always without labels. But this suggestion here should work and this is also how it was planned I think.